npm-cli/src/Pesy.re

```diff
@@ -36,188 +24,94 @@ let substituteTemplateValues = s =>
-let runTestsPath = Path.(testExeDir / testExeFileName);
-if (!exists(runTestsPath)) {
-  let runTests = runTestsTemplate |> substituteTemplateValues;
-
-  write(runTestsPath, runTests);
-};
-
-let readMePath = Path.(projectPath / "README.md");
-
-if (!exists(readMePath)) {
-  let readMe = readMeTemplate |> substituteTemplateValues;
-  write(readMePath, readMe);
-};
-
-let gitignorePath = Path.(projectPath / ".gitignore");
-
-if (!exists(gitignorePath)) {
-  let gitignore = gitignoreTemplate |> substituteTemplateValues;
-  write(".gitignore", gitignore);
-};
-
-let azurePipelinesPath = Path.(projectPath / "azure-pipelines.yml");
-
-if (!exists(azurePipelinesPath)) {
-  let esyBuildSteps = esyBuildStepsTemplate |> substituteTemplateValues;
-  copyTemplate(
-    Path.("azure-pipeline-templates" / "pesy-azure-pipelines.yml"),
-    Path.(projectPath / "azure-pipelines.yml"),
-  );
-  mkdirp(".ci");
-  let ciFilesPath = Path.(projectPath / ".ci");
-  write(Path.(ciFilesPath / "esy-build-steps.yml"), esyBuildSteps);
-  copyTemplate(
-    Path.("azure-pipeline-templates" / "pesy-publish-build-cache.yml"),
-    Path.(ciFilesPath / "publish-build-cache.yml"),
-  );
-  copyTemplate(
-    Path.("azure-pipeline-templates" / "pesy-restore-build-cache.yml"),
-    Path.(ciFilesPath / "restore-build-cache.yml"),
-  );
-};
-
-let libKebab = packageNameKebabSansScope;
-let duneProjectFile = Path.(projectPath / "dune-project");
-if (!exists(duneProjectFile)) {
-  write(duneProjectFile, spf({|(lang dune 1.2)
-(name %s)
-|}, libKebab));
-};
-
-let opamFileName = libKebab ++ ".opam";
-let opamFile = Path.(projectPath / opamFileName);
-if (!exists(opamFile)) {
-  write(opamFile, "");
-};
-
-let rootDuneFile = Path.(projectPath / "dune");
-
-if (!exists(rootDuneFile)) {
-  write(rootDuneFile, "(ignored_subdirs (node_modules))");
-};
-
-/* TODO: Box drawing characters are transform to weird char by bucklescript. Investigate */
-/* Printf.printf("\x1b[1m%s@%s\x1b[0m\n", packageNameKebabSansScope, version); */
-/* Utils.renderAscTree([ */
-/*   [ */
-/*     "executable", */
-/*     spf( */
-/*       "bin:    %sApp.re as %sApp.exe", */
-/*       packageNameUpperCamelCase, */
-/*       packageNameUpperCamelCase, */
-/*     ), */
-/*     spf( */
-/*       "require: [%s]", */
-/*       List.fold_left( */
-/*         (acc, e) => acc ++ " " ++ e, */
-/*         "", */
-/*         [spf("\"%s\"", packageLibName)], */
-/*       ), */
-/*     ), */
-/*   ], */
-/*   ["library", "require: []"], */
-/*   [ */
-/*     "test", */
-/*     spf( */
-/*       "bin:    Test%s.re as Test%s.exe", */
-/*       packageNameUpperCamelCase, */
-/*       packageNameUpperCamelCase, */
-/*     ), */
-/*     spf( */
-/*       "require: [%s]", */
-/*       String.concat(", ", [spf("\"%s\"", packageLibName)]), */
-/*     ), */
-/*   ], */
-/* ]); */
-/* print_newline(); */
+let substituteFileNames = s =>
+  s
+  |> Js.String.replaceByRe(
+       [%bs.re "/__PACKAGE_NAME_FULL__/g"],
```

Comment:
Why was this change in the template variable required? (`__PACKAGE_NAME_FULL__`)

Reply:
Windows couldn't handle files named `<PACKAGE_NAME_FULL>` so this is specific for file names, the in-file replaces are the same as before.